python/singa/utils.py

```diff
@@ -45,3 +46,395 @@ def update_progress(progress, info):
+        left,upper=0,(size[1]-size[0])/2
+        right,bottom =size[0],(size[1]+size[0])/2
+    elif position == 'bottom':
+        left,upper=0,size[1]-size[0]
+        right,bottom =size[0],size[1]
+    else:
+        raise Exception('position is wrong')
+    box =(left,upper,right,bottom) 
+    new_img = img.crop(box)
+
+    new_img = img.resize(crop)
+    #print box+crop
+    #print "crop to box %d,%d,%d,%d and scale to %d,%d" % (box+crop)
+    return new_img
+
+def do_resize(img,small_size):
+    size = img.size
+    if size[0]<size[1]:
+        new_size = ( small_size, int(small_size*size[1]/size[0]) )
+    else:
+        new_size = ( int(small_size*size[0]/size[1]), small_size )
+    new_img=img.resize(new_size)
+    #print 'resize to (%d,%d)' % new_size
+    return new_img
+ 
+
+def do_color_cast(img,scale):
+
+    x = np.asarray(img, dtype='uint8')
+    x.flags.writeable = True  
+    cast_value=[0,0,0]
+    for i in range(3):
+        r= random.randint(0,1)
+        if r:
+            cast_value[i] = random.randint(-scale,scale)
+    for w in range(img.size[0]):
+        for h in range(img.size[1]):
+            for c in range(3):
+                if cast_value[c]==0:
+                    continue
+                v=x[w][h][c]+cast_value[c] 
+                if v<0:
+                    v=0
+                if v>255:
+                    v=255
+                x[w][h][c]=v
+    new_img= Image.fromarray(x.astype('uint8'), 'RGB')
+    return new_img
+
+
+def do_enhance(img,scale):
+
+    # Color,Contrast,Brightness,Sharpness
+    enhance_value=[1.0,1.0,1.0,1.0]
+    for i in range(4):
+        r= random.randint(0,1)
+        if r:
+            enhance_value[i] = random.uniform(1-scale,1+scale)
+    if not enhance_value[0]==1.0:
+        enhancer = ImageEnhance.Color(img) 
+        img = enhancer.enhance(enhance_value[0])
+    if not enhance_value[1]==1.0:
+        enhancer = ImageEnhance.Contrast(img) 
+        img = enhancer.enhance(enhance_value[1])
+    if not enhance_value[2]==1.0:
+        enhancer = ImageEnhance.Brightness(img) 
+        img = enhancer.enhance(enhance_value[2])
+    if not enhance_value[3]==1.0:
+        enhancer = ImageEnhance.Sharpness(img) 
+        img = enhancer.enhance(enhance_value[3])
+    return img
+
+def do_flip(img):
+    #print 'flip'
+    new_img = img.transpose(Image.FLIP_LEFT_RIGHT)
+    return new_img     
+ 
+def get_list_sample(l,sample_size):
+    return [ l[i] for i in sorted(random.sample(xrange(len(l)), sample_size))]
+
+class Imgtool():
+
+    def __init__(self):
+        self.imgs=[]
+        return
+
+    def load(self,path):
+        img = load_img(path)
+        self.imgs=[img]
+        return self
+
+    def set(self,imgs):
+        self.imgs = imgs
+        return self
+
+    def append(self,img):
+        self.imgs.append(img)
+        return self
+
+    def get(self):
+        return self.imgs
+ 
+    def resize_by_range(self,(begin,end),step=1,k=1,update=True):
```

Comment:
don't use () in arg list.
[begin, end) -> begin, end or low, high or use rng. [ )  include begin, exclude end.
step is not necessary?